enhancements/kubelet/user-namespaces-support.md

```diff
@@ -0,0 +1,435 @@
+then the kube-apiserver may expect a pod is confined by a user namespace when it is not, and accidentally give it more privileges than it deserves.
+Thus, `UserNamespacesPodSecurityStandards` is currently in alpha, and will not advance to GA until three releases after Kubelet supports the `UserNamespacesSupport`
+feature gate.
+This feature gate is also the main feature gate used to toggle the SCC changes above. This is done because the same version skew issue exists in Openshift,
+and it would be better to keep UserNamespacesSupport on by default (not skewing from upstrem).
+
+ProcMountType is not directly related to user namespaces, but it does rely on them, and is useful for podman in Openshift pod use cases. Specifically,
+it's needed to allow podman to configure networking for the sub containers (as mounting a sub '/proc' requires the whole '/proc' be read-write, and '/proc' needs
+to be writable to configure sysctls)
+
+The goal is to graduate this feature in 4.18, meaning `UserNamespacesPodSecurityStandards` and `ProcMountType` will be enabled by default, and will not mark
+a cluster as TechPreviewNoUpgrade.
+
+### Risks and Mitigations
+
+- Allowing user namespaces does open Openshift users to theoretical kernel vulnerabilities
+    - While user namespaces have existed for a while, kernel concepts like idmapped mounting
+      have not, and the newness could be seen as risky.
+    - However, the tangible security advantages of allowing user namespaces outweigh the theoretical
+      security risks.
+    - Plus, for the majority of users, user namespaces will not be enabled to begin with.
+
+### Drawbacks
+
+## Open Questions [optional]
+
+## Test Plan
+
+- e2e tests for a user namespaced pod, especially with different volume types to verify kernel idmapped mount support
+- long-term: e2e tests for running podman in a pod, so we have an established test path for users to know what works and what doesn't.
+
+## Graduation Criteria
+
+### Dev Preview -> Tech Preview
+
+- All pieces of this enhancement implemented, within the TPNU feature set
+- Extensive documentation on enablement and common pitfalls
+- Unit and e2e coverage
+- Gather feedback from users rather than just developers (there are customers who are interested in trying this out)
+
+### Tech Preview -> GA
+
+- More testing (upgrade, downgrade, scale)
+- Sufficient time for feedback
+- Available by default
+- Backhaul SLI telemetry
+- Document SLOs for the component
+- Conduct load testing
+- User facing documentation created in [Openshift-docs](https://github.com/Openshift/Openshift-docs/)
+
+### Removing a deprecated feature
+N/A
+
+## Upgrade / Downgrade Strategy
+
+- To begin with, this feature will be gated by TechPreviewNoUpgrade, so it will not be able to upgrade
+- A downgrade of the apiserver will cause this feature to not work as well, as the SCC changes will be lost on older versions
```

Comment:
could you clarify this thinking?  It appears to me that when downgrading, the restrict-v3 will lose the new field and start matching pods with `hostUsers=true`

Reply:
do the feature gates persist across downgrades? I imagine the user isn't explicitly enabling user namespaces support (or they couldn't upgrade from 4.17), but I don't know what feature set continues to be enabled if you downgrade, but a feature gate qualifies the clusetr as TPNU (like in 4.18 if we GA user namespaces support ,that will be enabled, what happens to it on downgrade?)

Reply:
I'm also assuming SCCs get persisted across downgrades, is that correct?

Reply:
> I'm also assuming SCCs get persisted across downgrades, is that correct?

SCCs are persisted, but when the CRD schema is updated, reading `restricted-v3` will no longer have the field you're adding in this enhancement.

Reply:
reducing the list to `restricted-v3` (allowed by all by default) and `nested-container` (opted into by a namespace admin, which we'll add documentation for a cluster admin to comb for in case a downgrade is needed)